src/client/iframe.ts

```diff
@@ -0,0 +1,101 @@
+import { PluginDevMode, PluginClient, PluginOptions } from './client'
+import { Message, ModuleProfile } from '../types'
+import { ApiMap, getApiMap, listenOnThemeChanged } from './api'
+
+/**
+ * Check if the sender has the right origin
+ * @param msgOrigin The origin of the incoming message
+ * @param devMode Devmode options
+ */
+function checkOrigin(msgOrigin: string, devMode?: PluginDevMode) {
+  const localhost = devMode ? [
+    `http://127.0.0.1:${devMode.port}`,
+    `http://localhost:${devMode.port}`,
+    `https://127.0.0.1:${devMode.port}`,
+    `https://localhost:${devMode.port}`,
+  ] : []
+  return [
+    "http://remix-alpha.ethereum.org",
+    "http://remix.ethereum.org",
+    "https://remix-alpha.ethereum.org",
+    "https://remix.ethereum.org",
+    ...localhost
+  ].includes(msgOrigin)
+}
+
+/**
+ * Start listening on the IDE though PostMessage
+ * @param store A store to put the messages into
+ */
+function connectIframe(client: PluginClient) {
```

Comment:
as a first tentative, we could try to implement function `connectWS(client: PluginClient) {`